scipy/sparse/coo.py

```diff
@@ -278,31 +275,26 @@ def tocsc(self, copy=False):
             return csc_matrix(self.shape, dtype=self.dtype)
         else:
             M,N = self.shape
+            self.sum_duplicates()
             idx_dtype = get_index_dtype((self.col, self.row),
                                         maxval=max(self.nnz, M))
+            row = self.row.astype(idx_dtype, copy=False)
+            col = self.col.astype(idx_dtype, copy=False)
```

Comment:
However, computing idx_dtype proably is necessary since it's affected by nnz, so it should be used to determine the out array dtypes.